### PR TITLE
Add an option to filter hom_ref_calls.

### PR DESCRIPTION
**Pull request by tneymanov**

### CODE REVIEW

**Review by samanvp — COMMENTED** [review]
I had this comments last week, sorry, forgot to sent them.
They might be out of date with respect to your last commit.

File: gcp_variant_transforms/beam_io/vcf_parser.py [hunk 1]
```diff
@@ -637,21 +646,29 @@ def _lookup_encoded_sample_name(self, sample_name):
...
+      moved_to_hom_ref = False
       for (key, value) in list(sample.items()):
         if key == GENOTYPE_FORMAT_KEY:
           if isinstance(value, tuple):
             genotype = []
             for elem in value:
               genotype.append(MISSING_GENOTYPE_VALUE if elem is None else elem)
+            if self._move_hom_ref_calls and not any(genotype):
+              moved_to_hom_ref = True
+              break
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/beam_io/vcf_parser.py:665]: But we need to have them both checked - if GT is 0 or if GT is (0,0,...,0) - both cases should be handled. Also,  switching Nones to MISSING_GENOTYPE_VALUE is vital to make this work:

```
>>> a = (0,0,0)
>>> b = (0,1,0)
>>> c = (0,0,None)
>>> any(a)
False
>>> any(b)
True
>>> any(c)
False
```

File: gcp_variant_transforms/beam_io/vcfio.py [hunk 2]
```diff
@@ -271,13 +275,16 @@ def __init__(self,
         to deal with same sample_name used across multiple VCF files.
       use_1_based_coordinate: specify whether the coordinates should be stored
         in BQ using 0-based exclusive (default) or 1-based inclusive coordinate.
+      move_hom_ref_calls: If true, filter out 0 GT data out of call list and add
+        the call name to a hom_ref_calls column.
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/beam_io/vcfio.py:279]: Done.

File: gcp_variant_transforms/beam_io/vcfio.py [hunk 3]
```diff
@@ -347,6 +356,8 @@ def __init__(
...
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/beam_io/vcfio.py:360]: Done.

File: gcp_variant_transforms/beam_io/vcfio.py [hunk 4]
```diff
@@ -425,6 +443,8 @@ def __init__(
...
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/beam_io/vcfio.py:447]: Done.

File: gcp_variant_transforms/options/variant_transform_options.py [hunk 5]
```diff
@@ -221,6 +221,11 @@ def add_arguments(self, parser):
         type='bool', default=False, nargs='?', const=True,
         help=('Add raw sample name (from VCF Header line) as a subfield under '
               'the call column in addition to sample_id.'))
+    parser.add_argument(
+        '--move_hom_ref_calls',
+        type='bool', default=False, nargs='?', const=True,
+        help=('Filter out calls with unchanged GT values into hom_ref_calls '
+              'column.'))
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/options/variant_transform_options.py:228]: This is really weird, I added more text there to imply non-reversibility of the flow, but don't see them...

Anyways, Done.

**Review by samanvp — APPROVED** [review]
Thanks!

**Pull request merged**